trimpairs.py

```diff
@@ -3,72 +3,111 @@
 languages with coordinates in the tsv file
 """
 
-def filterPairs(filename, languages):
-    pairFile = open(filename, "r")
+import json
+
+
+def filter_pairs(filename, languages, alt_codes):
+    """
+    Filters out any language that is not in
+    the list of valid languages
+    """
+
+    pair_file = open(filename, "r")
     filtered = []
-    count = 1
-    for line in pairFile:
+    for line in pair_file:
         line = line.strip().split()
-        lang1 = line[3][1:4]
-        lang2 = line[1][1:4]
+
+        lang1 = line[3][1:-2]
+        lang2 = line[1][1:-2]
+
         if lang1 in languages and lang2 in languages:
-            #getting rid of last comma
+            # Getting rid of last comma
             line[-1] = line[-1][:-1]
             joined = " ".join(line)
             filtered.append(joined)
 
-    pairFile.close()
+        elif (
+                lang1 in alt_codes and alt_codes[lang1] in languages
+        ) and (
+            lang2 in alt_codes and alt_codes[lang2] in languages
+        ):
+            # Getting rid of last comma
+            line[-1] = line[-1][:-1]
+            line[1] = line[1][0] + alt_codes[lang2] + line[1][-2:]
+            line[3] = line[3][0] + alt_codes[lang1] + line[3][-2:]
+            joined = " ".join(line)
+            filtered.append(joined)
+
+    pair_file.close()
     return filtered
 
-def getLanguagesWithCoords(filename):
 
-    langFile = open(filename, "r")
-    langDict = {}
-    for line in langFile:
+def get_languages_with_coords(filename):
+    """
+    Gets the languages with their coordinates
+    on the Earth
+    """
+
+    lang_file = open(filename, "r")
+    lang_dict = {}
+    for line in lang_file:
         line = line.strip().split(",")
         lat, lon = float(line[1]), float(line[2])
-        langDict[line[0]] = [lon, lat]
-    langFile.close()
-    return langDict
+        lang_dict[line[0]] = [lon, lat]
+    lang_file.close()
+    return lang_dict
+
+
+def get_alternate_codes(filename):
+    """
+    Reads the alternate codes for languages from file
+    and then returns a dict of that data
+    """
+
+    code_file = open(filename, 'r')
+    code_str = code_file.read()
+    code_file.close()
+    return json.loads(code_str)
 
 if __name__ == "__main__":
-    langDict = getLanguagesWithCoords("apertium-languages.tsv")
-    filtered = filterPairs("pairs.json.txt", langDict)
-    apertiumFile = open("apertiumPairs.json", "w")
+    LANG_DICT = get_languages_with_coords("apertium-languages.tsv")
+    ALT_CODES = get_alternate_codes('codes.json')
+    FILTERED = filter_pairs("pairs.json.txt", LANG_DICT, ALT_CODES)
+    APERTIUM_FILE = open("apertiumPairs.json", "w")
 
-    apertiumFile.write("{\n")
-    apertiumFile.write('"type": "FeatureCollection",\n')
-    apertiumFile.write('"pairs": [\n')
-    for line in filtered[:-1]:
-        apertiumFile.write("\t"+line+"\n")
-        apertiumFile.write(",\n")
+    APERTIUM_FILE.write("{\n")
```

Comment:
Why did you rename these variables?

Reply:
If there's a good reason to have changed them, that's fine.  I'm just curious what the reason was.